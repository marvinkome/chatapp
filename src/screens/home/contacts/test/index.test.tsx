import * as React from 'react';
import { shallow } from 'enzyme';
import { Contacts } from '../index';
import View from '../view';

describe('<Contacts/> listing tab', () => {
    const lists = [
        {
            name: 'Larry John',
            about: 'I am mocked person :)',
            image: ''
        }
    ];

    // snapshot
    it('renders correctly', () => {
        const app = shallow(<Contacts />);

        expect(app).toMatchSnapshot();
    });

    // list all chats
    it('should list all chats', () => {
        const wrapper = shallow(<View />);
        const list = wrapper.find('List');

        expect(list).toMatchSnapshot();
        expect(list.props().children).toHaveLength(2);
    });
});
