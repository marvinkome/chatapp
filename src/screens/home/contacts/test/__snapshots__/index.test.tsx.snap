// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contacts/> listing tab renders a contact 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Listing
    listItem={
      Object {
        "about": "I am mocked person :)",
        "image": "",
        "name": "Larry John",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "avatar": "",
      "subtitle": "I am mocked person :)",
      "title": "Larry John",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "avatar": "",
        "subtitle": "I am mocked person :)",
        "title": "Larry John",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Contacts/> listing tab renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contacts />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Contacts/> listing tab should list all chats 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): ShallowWrapper {
    "length": 1,
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ScreenView />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <List
          containerStyle={
            Object {
              "borderTopWidth": 0,
              "marginTop": 13,
            }
          }
        >
          <Listing
            listItem={
              Object {
                "about": "Just a cool guy",
                "image": Object {
                  "testUri": "../../../static/pp.jpg",
                },
                "name": "Larry John",
              }
            }
          />
          <Listing
            listItem={
              Object {
                "about": "Currently coding",
                "image": Object {
                  "testUri": "../../../static/yuna.jpg",
                },
                "name": "Jane Doe",
              }
            }
          />
        </List>,
        "style": Object {
          "backgroundColor": "#fff",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Listing
              listItem={
                Object {
                  "about": "Just a cool guy",
                  "image": Object {
                    "testUri": "../../../static/pp.jpg",
                  },
                  "name": "Larry John",
                }
              }
            />,
            <Listing
              listItem={
                Object {
                  "about": "Currently coding",
                  "image": Object {
                    "testUri": "../../../static/yuna.jpg",
                  },
                  "name": "Jane Doe",
                }
              }
            />,
          ],
          "containerStyle": Object {
            "borderTopWidth": 0,
            "marginTop": 13,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "listItem": Object {
                "about": "Just a cool guy",
                "image": Object {
                  "testUri": "../../../static/pp.jpg",
                },
                "name": "Larry John",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "listItem": Object {
                "about": "Currently coding",
                "image": Object {
                  "testUri": "../../../static/yuna.jpg",
                },
                "name": "Jane Doe",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <List
            containerStyle={
              Object {
                "borderTopWidth": 0,
                "marginTop": 13,
              }
            }
          >
            <Listing
              listItem={
                Object {
                  "about": "Just a cool guy",
                  "image": Object {
                    "testUri": "../../../static/pp.jpg",
                  },
                  "name": "Larry John",
                }
              }
            />
            <Listing
              listItem={
                Object {
                  "about": "Currently coding",
                  "image": Object {
                    "testUri": "../../../static/yuna.jpg",
                  },
                  "name": "Jane Doe",
                }
              }
            />
          </List>,
          "style": Object {
            "backgroundColor": "#fff",
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Listing
                listItem={
                  Object {
                    "about": "Just a cool guy",
                    "image": Object {
                      "testUri": "../../../static/pp.jpg",
                    },
                    "name": "Larry John",
                  }
                }
              />,
              <Listing
                listItem={
                  Object {
                    "about": "Currently coding",
                    "image": Object {
                      "testUri": "../../../static/yuna.jpg",
                    },
                    "name": "Jane Doe",
                  }
                }
              />,
            ],
            "containerStyle": Object {
              "borderTopWidth": 0,
              "marginTop": 13,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "listItem": Object {
                  "about": "Just a cool guy",
                  "image": Object {
                    "testUri": "../../../static/pp.jpg",
                  },
                  "name": "Larry John",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "listItem": Object {
                  "about": "Currently coding",
                  "image": Object {
                    "testUri": "../../../static/yuna.jpg",
                  },
                  "name": "Jane Doe",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Listing
          listItem={
            Object {
              "about": "Just a cool guy",
              "image": Object {
                "testUri": "../../../static/pp.jpg",
              },
              "name": "Larry John",
            }
          }
        />,
        <Listing
          listItem={
            Object {
              "about": "Currently coding",
              "image": Object {
                "testUri": "../../../static/yuna.jpg",
              },
              "name": "Jane Doe",
            }
          }
        />,
      ],
      "containerStyle": Object {
        "borderTopWidth": 0,
        "marginTop": 13,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "listItem": Object {
            "about": "Just a cool guy",
            "image": Object {
              "testUri": "../../../static/pp.jpg",
            },
            "name": "Larry John",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "listItem": Object {
            "about": "Currently coding",
            "image": Object {
              "testUri": "../../../static/yuna.jpg",
            },
            "name": "Jane Doe",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Listing
            listItem={
              Object {
                "about": "Just a cool guy",
                "image": Object {
                  "testUri": "../../../static/pp.jpg",
                },
                "name": "Larry John",
              }
            }
          />,
          <Listing
            listItem={
              Object {
                "about": "Currently coding",
                "image": Object {
                  "testUri": "../../../static/yuna.jpg",
                },
                "name": "Jane Doe",
              }
            }
          />,
        ],
        "containerStyle": Object {
          "borderTopWidth": 0,
          "marginTop": 13,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "listItem": Object {
              "about": "Just a cool guy",
              "image": Object {
                "testUri": "../../../static/pp.jpg",
              },
              "name": "Larry John",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "listItem": Object {
              "about": "Currently coding",
              "image": Object {
                "testUri": "../../../static/yuna.jpg",
              },
              "name": "Jane Doe",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
